src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/data/EnvelopeRepositoryTest.java

```diff
@@ -112,6 +112,30 @@ void should_mark_existing_envelope_as_deleted() {
         assertThat(envelopeAfterUpdate).hasValueSatisfying(env -> assertThat(env.isDeleted).isEqualTo(true));
     }
 
+    @Test
+    void should_update_envelope_as_notification_sent() {
+        // given
+        var newEnvelope = new NewEnvelope(
+            "container",
+            "hello.zip",
+            now(),
+            now().plusSeconds(100),
+            Status.REJECTED
+        );
+
+        UUID id = repo.insert(newEnvelope);
+        Optional<Envelope> envelope = repo.find(id);
+        assertThat(envelope).hasValueSatisfying(env -> assertThat(env.pendingNotification).isEqualTo(false));
+
+        // when
+        int updateCount = repo.updatePendingNotification(id);
+        Optional<Envelope> envelopeAfterUpdate = repo.find(id);
+
+        // then
+        assertThat(updateCount).isEqualTo(1);
+        assertThat(envelopeAfterUpdate).hasValueSatisfying(env -> assertThat(env.pendingNotification).isEqualTo(true));
```

Comment:
For single assertion below should be fine in my opinion but up to you
```
assertThat(envelopeAfterUpdate.get(0).pendingNotification).isEqualTo(true);

```

Reply:
removed the additional assertion